src/widgets/table.rs

```diff
@@ -399,13 +399,22 @@ impl<'a> Table<'a> {
     /// Get all offsets and widths of all user specified columns
     /// Returns (x, width)
     fn get_columns_widths(&self, max_width: u16, selection_width: u16) -> Vec<(u16, u16)> {
-        let mut constraints = Vec::with_capacity(self.widths.len() * 2 + 1);
+        // Default to equal column widths if not provided
+        let widths = if self.widths.is_empty() {
+            let max_col_count = self.rows.iter().map(|r| r.cells.len()).max();
+            let header_col_count = self.header.as_ref().unwrap_or(&Row::default()).cells.len();
+            let col_count = max_col_count.unwrap_or(header_col_count);
+            vec![Constraint::Percentage(100 / col_count as u16); col_count]
```

Comment:
One problem with using Percentage / Ratio is that it's a percentage of the entire space, not a percentage of the leftover space after allocations. This means this code will always over allocate space to the columns and have to choose one or more constraint to break. This can play weird with rendering expectations especially for small columns.

Reply:
Oh ok, interesting. Another option might be `Constraint::Length(max_width / col_count - col_count + 1)` in that case?

Reply:
On second thought, don't think this will work.